functions/rename-requirement-and-slot-name.js

```diff
@@ -0,0 +1,109 @@
+// General script to rename the requirement or slot name in Firestore if it
+// has been changed in the requirements data.
+const admin = require('firebase-admin');
+const fs = require('fs');
+const serviceAccount = require('./service-account.json');
+
+admin.initializeApp({
+  credential: admin.credential.cert(serviceAccount),
+  databaseURL: 'https://cornelldti-courseplan-dev.firebaseio.com',
+});
+
+const db = admin.firestore();
+const userOverriddenReqsCollection = db.collection('user-overridden-reqs');
+const userOnboardingData = db.collection('user-onboarding-data');
+
+/** The current name of the requirement or slot name that we want to change. */
+const oldName;
+/** The name that we want to change the requirement or slot name to. */
+const newName;
+/** True if the name we are changing is the name of a slot.
+ * False if the name we are changing is the name of a requirement. */
+const isChangingSlotName;
+
+/** updates the old name of the requirement or slot name with the new name in
+ * userOverriddenReqsCollection
+ */
+const updateNames = doc => {
+  const newDoc = {};
+  for (const [key, value] of Object.entries(doc)) {
+    const nameInDB = isChangingSlotName ? key[1] : key[0];
+    if (nameInDB === oldName) {
+      const newKey = isChangingSlotName ? [key[0], newName] : [newName, key[1]];
+      newDoc[newKey] = value;
```

Comment:
Not sure whether it will work. You are indexing a array with an array.

Reply:
@SamChou19815 I believe in this script we're assuming that the key for `newDoc` is an array `[requirement name, slot name]`, so the idea is to set the value of `newDoc` (an object) at `newKey`.

Reply:
I don't think that will work on JS. You can only use string or number to index things.